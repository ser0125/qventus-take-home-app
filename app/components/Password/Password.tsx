"use client";
import { ChangeEvent, useEffect, useState } from "react";
import Input from "../Input/Input";
import CheckValidation from "../CheckValidation/CheckValidation";
import useValidation from "@/app/hooks/useValidation";
import { PasswordReq } from "@/app/interfaces/PasswordReq";

const Password = ({ passwordReqs }: { passwordReqs: PasswordReq[] }) => {
  const { handleChange, arrayChecked } = useValidation(passwordReqs);
  return (
    <div className="flex flex-col md:flex-row items-center pt-5">
      <Input onChange={handleChange} />
      <div className="flex flex-col mt-3">
        {arrayChecked &&
          arrayChecked.map((req: any) => (
            <CheckValidation
              key={req.id}
              label={req.label}
              isValid={req.isValid}
            />
          ))}
      </div>
    </div>
  );
};

export default Password;
