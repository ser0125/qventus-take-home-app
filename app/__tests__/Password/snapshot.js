import { render } from "@testing-library/react";
import Password from "../../components/Password/Password";

it("renders password unchanged", () => {
  const { container } = render(<Password />);
  expect(container).toMatchSnapshot();
});
