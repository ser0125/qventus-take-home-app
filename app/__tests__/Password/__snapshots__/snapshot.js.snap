// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders password unchanged 1`] = `
<div>
  <div
    class="flex items-center pt-5"
  >
    <input
      class="h-7 border border-current focus:outline-blue-500 focus:shadow-md"
      type="text"
    />
    <div
      class="flex flex-col"
    />
  </div>
</div>
`;
